### PR TITLE
Report builder creation warnings according to compiler args

### LINKED ISSUE #4030: Type check errors contain internal options warnings when --nowarn:75 is passed
Warnings about internal options of the following form are returned in `FSharpCheckFileResults` and `FSharpCheckProjectResults` even when `--nowarn:75` is passed in `FSharpProjectOptions.OtherOptions`. These errors have error number 75.
```
commandLineArgs (0,1)-(0,1) parameter warning The command-line option 'times' is for test purposes only
```

In VS these warnings are ignored in the code below and after a quick search I don't see other places where these errors are shown to a user. I suppose it is a bug and these errors should not appear in the first place.
https://github.com/Microsoft/visualfsharp/blob/716b8e9998c1d764ba23dc37602e77b506dcfc64/vsintegration/src/FSharp.Editor/Diagnostics/DocumentDiagnosticAnalyzer.fs#L80-L84

### PR DESCRIPTION
**Pull request by auduchinok**
Fixes #4030.

### CODE REVIEW

**Comment by auduchinok** [comment]
It turned out CompilationErrorLogger didn't change error severities according to WarnAsWarn or WarnAsError options. I fixed it and now I'm waiting for test results to check whether anything needs to be fixed.

**Comment by auduchinok** [comment]
@saul Done.

**Comment by saul** [comment]
Cheers! Looks great

File: tests/service/ProjectAnalysisTests.fs [hunk 1]
```diff
@@ -5195,3 +5195,16 @@ type A(i:int) =
 
     | Some decl -> failwithf "unexpected declaration %A" decl
     | None -> failwith "declaration list is empty"
+
+
+[<TestCase(([||]: string[]), ([||]: bool[]))>]
```
Comment (dsyme) [inline tests/service/ProjectAnalysisTests.fs]: This isn't urgent, but just to say that I personally never use ``TestCase`` attributes in the F# tests, and prefer a loop over an F# list of test case data inputs.  Indeed I usually remove any uses of ``TestCase`` when I come across them later

While using TestCase attribute has some advantages (e.g. they get reported as separate tests in a test explorer), I believe that just using F# code to program the space of test cases is more uniform and in the long run easier to understand - and encourages people to be really expansive in how they meta-program up a very broad range of systematic test cases.

Comment (dsyme) [reply tests/service/ProjectAnalysisTests.fs]: (Programming up the test case matrix with F# data is also more typesafe of course)

**Comment by KevinRansom** [comment]
Thanks for this.

**Pull request merged**